server/src/main/java/org/elasticsearch/script/AbstractSortScript.java

```diff
@@ -29,14 +31,25 @@
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Function;
 
 abstract class AbstractSortScript implements ScorerAware {
 
-    private static final Map<String, String> DEPRECATIONS = Map.of(
-            "doc",
-            "Accessing variable [doc] via [params.doc] from within a sort-script is deprecated in favor of directly accessing [doc].",
-            "_doc",
-            "Accessing variable [doc] via [params._doc] from within a sort-script is deprecated in favor of directly accessing [doc].");
+    private static final DeprecationLogger deprecationLogger =
+            new DeprecationLogger(LogManager.getLogger(DynamicMap.class));
+    private static final Map<String, Function<Object, Object>> FUNCTIONS = Map.of(
```

Comment:
I might name this like `PARAMS_FUNCTIONS` or something like that, so it is clear they are part of `params`.

Reply:
Changed.